### PR TITLE
Now read-bioinfo-metadata splits data and extract files for each batch

### PR DESCRIPTION
**Pull request by Shettland**
In-depth description:

- Now before starting with data processing, `j_data` is separated into smaller copies, according to `r1_fastq_filepath` unique values found in the samples and all the process is iterated over each of these set of samples.
- `output_folder` is now a mandatory arg to include in each method called by the auxiliar script (in this case `viralrecon.py`). This was needed in order to extract `long_table*.json ` file to its respective batch folder
- `handling_tables()` is a now separated from `handling_files()` as an independent method, just for readability
- Changed file pattern matching to `sequencing_sample_id` as now `pipeline_manager` names files using that field.
- Each validated file is now extracted during `add_bioinfo_files_path()`
- Included two new fields in `bioinfo_config.json`: "extract" and "split_by_batch"
- `"extract"` should be set to true for those files in the config key that need to be copied to the original folder with samples
- `"split_by_batch"` indicates those tables that have information for all the samples, and should be split into smaller copies of the table only with the samples for each batch into their respective folder.
- These two fields are used as markers for the new methods: `extract_file()` and `split_table_by_batch()`

Other changes:
- Now a json file with process logs is created using `LogSum` class from `log_summary.py`
- Reduced scann_directory() time consumption by simplifying os.walk() to be executed only one time
- `read_csv_file_return_dict` from `utils.py` now tries to detect separator using file extension if no `sep` argument is given

`test_modules.yml `workflow actions will fail until we update the version for `upload-artifact` used in them. I'll include it in the next PR

### CODE REVIEW

**Review by Daniel-VM — COMMENTED** [review]
Those are great enhancements 
just few comments to address and you'll be ready to go 🚀

**Comment by Daniel-VM** [comment]
Don't forget to update CHANGELOG 🙏🏾

File: relecov_tools/conf/bioinfo_config.json [hunk 1]
```diff
@@ -26,6 +27,7 @@
             "header_row_idx": 1,
             "required": false,
             "function": "handle_pangolin_data",
+            "extract": true,
```
[the remark this answers is not in the document]
Comment (Shettland) [reply relecov_tools/conf/bioinfo_config.json:30]: I agree. My idea was to include them inside README right below module description, the same way as for build-schema

File: relecov_tools/read_bioinfo_metadata.py [hunk 2]
```diff
@@ -270,8 +285,45 @@ def add_bioinfo_results_metadata(self, files_dict, j_data):
...
+        extdict = {".csv": ",", ".tsv": "\t", ".tab": "\t"}
+        if file_ext in extdict.keys():
+            data = relecov_tools.utils.read_csv_file_return_dict(
+                file_name=file_list[0],
+                sep=extdict.get(file_ext),
+                key_position=sample_idx_colpos,
+            )
+            return data
+        elif conf_tab_name.endswith(".gz"):
+            data = {}
```
[the remark this answers is not in the document]
Comment (Shettland) [reply relecov_tools/read_bioinfo_metadata.py:313]: I'll include it, but that won't stop the "missing metadata" message from arising, you'll just have two warning messages

[the remark this answers is not in the document]
Comment (Daniel-VM) [reply relecov_tools/read_bioinfo_metadata.py:318]: don't worry, that will be more informative. Future releases might include a method to untar gz files.

**Review by Daniel-VM — APPROVED** [review]
LGTM

**Pull request merged**